structure/structure.py

```diff
@@ -1,25 +1,64 @@
+import os
 import sys
 
+from dotenv import load_dotenv
+
+from griptape.artifacts import TextArtifact
 from griptape.drivers import (
     GriptapeCloudEventListenerDriver,
 )
 from griptape.events import (
     EventListener,
+    FinishStructureRunEvent,
 )
 from griptape.structures import Agent
 from griptape.tools import Calculator
 
+
+# Returns True if this program is being run from within Griptape Cloud or via the Skatepark emulator
+def is_running_in_managed_environment():
+    return "GT_CLOUD_RUN_ID" in os.environ
+
+
+# If running completely local, such as within an IDE, load environment vars.
+if not is_running_in_managed_environment():
+    load_dotenv()
+
 input = sys.argv[1]
 
+# We need an event driver to communicate events from our program back to Skatepark
+listener_base = os.environ.get("GRIPTAPE_SKATEPARK_URL_OVERRIDE", "http://127.0.0.1")
+listener_port = os.environ.get("GRIPTAPE_SKATEPARK_PORT_OVERRIDE", "5000")
```

Comment:
@SavagePencil late to the party on this sorry. What does the user's path to non-Skatepark runtime look like? They need to provide a real `base_url` (`https://cloud.griptape.ai`), and real API key. I think we should set up the code to make that switch as easy as possible, right now it feels kind of clunky.

Reply:
ugh you're right, and I don't want to do a ton of "if pure local do this, if Skatepark do that, if Cloud do this other thing." Let's start a thread to discuss.

Reply:
Totally - so glad to hear that. :)